File: src/main/java/facts/stmt/SkipStmt.java
package facts.stmt;

import astnodes.exp.Expr;
import java.util.ArrayList;
import java.util.List;

/**
 * No instruction fact
 */
public class SkipStmt extends Stmt {

    public SkipStmt(String pc) {
        super(pc);
    }

    @Override
    public String toString() {
        return (String.format("%sskip;", getLabel()));
    }

    @Override
    public List<Expr> getChildren() {
        return new ArrayList<>();
    }

    @Override
    public void replace(Expr oldExp, Expr newExp) {}
}

File: src/main/scala/astnodes/stmt/SkipStmt.scala
package astnodes.stmt

import astnodes.exp.Expr
import astnodes.exp.variable.Variable

/** No instruction fact
  */
case class SkipStmt(override val pc: String) extends Stmt(pc) {
  override def toString: String = String.format("%sskip;", labelString)
  override def subst(v: Variable, w: Variable): Stmt = this

  //override def withVisibleLabel: Stmt = copy(labelVisible = true)
}
